app/main.py

```diff
@@ -97,37 +99,70 @@ def grab_secrets(self):
-                _logger.info('Retrieving cert name:%s with version: %s', key_name, key_version)
+                # only cert_filename is needed, key_filename is ignored with _
+                key_name, key_version, cert_filename, _ = self._split_keyinfo(key_info)
+                _logger.info('Retrieving cert name:%s with version: %s output certFileName: %s', key_name, key_version, cert_filename)
                 cert = client.get_certificate(vault_base_url, key_name, key_version)
-                output_path = os.path.join(self._certs_output_folder, key_name)
+                output_path = os.path.join(self._certs_output_folder, cert_filename)
                 _logger.info('Dumping cert value to: %s', output_path)
                 with open(output_path, 'w') as cert_file:
                     cert_file.write(self._cert_to_pem(cert.cer))
 
-    def _dump_pfx(self, pfx, name):
+    def _dump_pfx(self, pfx, cert_filename, key_filename):
         import base64
         from OpenSSL import crypto
         p12 = crypto.load_pkcs12(base64.decodestring(pfx))
         pk = crypto.dump_privatekey(crypto.FILETYPE_PEM, p12.get_privatekey())
         cert = crypto.dump_certificate(crypto.FILETYPE_PEM, p12.get_certificate())
-        with open(os.path.join(self._keys_output_folder, name), 'w') as key_file:
-            key_file.write(pk)
-        with open(os.path.join(self._certs_output_folder, name), 'w') as cert_file:
-            cert_file.write(cert)
+
+        if (cert_filename == key_filename):
+            with open(os.path.join(self._keys_output_folder, key_filename), 'w') as key_file:                
+                _logger.info('Dumping key value to: %s', os.path.join(self._keys_output_folder, key_filename))
+                key_file.write(pk)
+
+            with open(os.path.join(self._certs_output_folder, cert_filename), 'w') as cert_file:
+                _logger.info('Dumping key value to: %s', os.path.join(self._certs_output_folder, cert_filename))
+                cert_file.write(cert)                
+        else:
+            # write to certs_keys folder when cert_filename and key_filename specified
+            with open(os.path.join(self._cert_keys_output_folder, key_filename), 'w') as key_file:
+                _logger.info('Dumping key value to: %s', os.path.join(self._keys_output_folder, key_filename))
+                key_file.write(pk)
+
+            with open(os.path.join(self._cert_keys_output_folder, cert_filename), 'w') as cert_file:
+                _logger.info('Dumping key value to: %s', os.path.join(self._cert_keys_output_folder, cert_filename))
+                cert_file.write(cert)    
+                    
+    @staticmethod
+    def _split_keyinfo(key_info):
+        key_parts = key_info.strip().split(':')
+        key_name = key_parts[0]
+        key_version = '' if len(key_parts) < 2 else key_parts[1]
+        cert_filename = key_name if len(key_parts) < 3 else key_parts[2]
```

Comment:
if the secret isn't backing a cert, than shouldn't it be just `file_name` instead of `cert_filename`?

Reply:
The naming is confusing - I think it would help if it was "filename" and then "key_filename" since the "cert_filename" is used for the simple "secret" case (not backed by a cert/key pair).

Also, there is a question of if we should error if you try to provide 2 filenames for a secret that can only use one of them.  (Or if you provide more than 2)

Reply:
This now sets the rule that only certs (with cert and key) are optionally renamed.  That makes a for a clearer behavior.

Reply:
@hexadite-idan For base case(secret has no cert and key) secret will have same name

Reply:
@raghu017 why would we support only cert renaming?

Reply:
My requirement is cert and key filenames. If there is a requirement for secret file naming it can be added by a dev later.